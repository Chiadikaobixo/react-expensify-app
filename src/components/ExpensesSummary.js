import React from "react";
import { connect } from 'react-redux'
import getVisibleExpenses from '../selectors/expenses'
import expenseTotal from "../selectors/expenseTotal";
import { Link } from 'react-router-dom'
import numeral from 'numeral'

export const ExpensesSummary = ({ count, total }) => {
    const expenseWord = count === 1 ? 'expense' : 'expenses'
    const formatedTotal = numeral(total / 100).format('$0,0.00')
    return (
        <div className="page-header">
            <div className="content-container">
                <h1
                    className="page-header__title">viewing <span>{count}</span>
                    {expenseWord} totalling <span>{formatedTotal}</span>
                </h1>
                <div className="page-header__actions">
                    <Link className="button" to="/create">Add Expense</Link>
                </div>
            </div>
        </div>
    )
}

const mapStateToProps = (state) => {
    const visibleExpense = getVisibleExpenses(state.expenses, state.filters)
    return {
        count: visibleExpense.length,
        total: expenseTotal(visibleExpense)
    }
}
export default connect(mapStateToProps)(ExpensesSummary)