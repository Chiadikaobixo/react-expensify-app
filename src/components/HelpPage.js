import React from "react";

const HelpPage = () => (
    <div>
      this is the HelpPage!!
    </div>
)

export default HelpPage