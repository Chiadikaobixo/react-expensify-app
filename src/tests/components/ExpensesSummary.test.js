import React from "react";
import {shallow} from 'enzyme'
import {ExpensesSummary} from "../../components/ExpensesSummary";

test('should correctly render Expenses summary with 1 Expense', () => {
    const wrapper = shallow(<ExpensesSummary count={1} total={234} />)
    expect(wrapper).toMatchSnapshot()
})

test('should correctly render Expenses summary with multiple Expenses', () => {
    const wrapper = shallow(<ExpensesSummary count={56} total={2347634190} />)
    expect(wrapper).toMatchSnapshot()
})