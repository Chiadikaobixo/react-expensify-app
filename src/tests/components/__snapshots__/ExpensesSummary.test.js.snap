// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render Expenses summary with 1 Expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      viewing 
      <span>
        1
      </span>
      expense
       totalling 
      <span>
        $2.34
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should correctly render Expenses summary with multiple Expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      viewing 
      <span>
        56
      </span>
      expenses
       totalling 
      <span>
        $23,476,341.90
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
